### ISSUE TITLE
Change configuration of sensors for live import and export data to use in Energy dashboard

### ISSUE THREAD

**Issue by jacopobac** [CJNE/ha-myenergi#180]
**Is your feature request related to a problem? Please describe.**
I would like to see on the Energy dashboard the actual generation, import and export of my home and PV system using the CT clamps of MyEnergi devices

**Describe the solution you'd like**
I'd like to use the sensors "sensor.myenergi_my_home_power_generation" and "sensor.myenergi_my_home_power_grid" in the energy hub in Home Assistant, but I can only select for import and export the sensors regarding total amount of energy imported and exported, not live data.

**Describe alternatives you've considered**
I am currently using the total amount, but the data is all messy.

**Additional context**
Here are the sensors I can choose from and the energy dashboard:

![image](https://user-images.githubusercontent.com/2984571/177639646-b3f171da-a6e6-4484-90dc-ac771fece2cb.png)

![image](https://user-images.githubusercontent.com/2984571/177639797-aa99504b-f411-4f4c-abcd-b4b1b743efec.png)

![image](https://user-images.githubusercontent.com/2984571/177639893-c7038079-0a12-4e81-8fdc-a715010fd029.png)

Thank you so much! I love this project!

**Comment by G6EJD** [comment]
I do this, no change needed, I kept the HA Energy dashboard, copied it to one called MyEnergi then changed the energy display entities to those from MyEnergy, it’s really simple to do.

**Comment by whistlebare** [comment]
I too would like to be able to see an entity for my Solar live generation rather than just a total of today

**Comment by LeiChat** [comment]
> I too would like to be able to see an entity for my Solar live generation rather than just a total of today

Unless I am mistaken, this is already included in the integration.
sensor.myenergi_home_power_generation

![image](https://github.com/CJNE/ha-myenergi/assets/23559551/7fd0e250-baf6-4f9b-b293-226dcc4d525d)

Does your master myenergi device have a CT connected and configured to measure PV generation? Without that, myenergi won't be capturing the power to their cloud and the ha-myenergi integration would not be able to extract the data using the myenergi API.

**Comment by whistlebare** [comment]
> > I too would like to be able to see an entity for my Solar live generation rather than just a total of today
> 
> Unless I am mistaken, this is already included in the integration. sensor.myenergi_home_power_generation
> 
> ![image](https://user-images.githubusercontent.com/23559551/261800866-7fd0e250-baf6-4f9b-b293-226dcc4d525d.png)
> 
> Does your master myenergi device have a CT connected and configured to measure PV generation? Without that, myenergi won't be capturing the power to their cloud and the ha-myenergi integration would not be able to extract the data using the myenergi API.

My issue is that I can't choose that entity within the 'Energy' tab in HA.  So when I go the the Add solar generation, I can't choose that entity

**Comment by G6EJD** [comment]
But that’s a HA issue not for this Integration.

**Comment by whistlebare** [comment]
But it says that if the entity is not there then contact the developer of the integration and ask them if they can    implement it...
  

  

  

  
  
  
  
  
>

**Comment by whistlebare** [comment]
![Screenshot_2023-08-19-15-22-11-38_c3a231c25ed346e59462e84656a70e50](https://github.com/CJNE/ha-myenergi/assets/126982604/89f63f4d-77b4-4911-87b5-6d80a4397fbe)

**Comment by whistlebare** [comment]
![Screenshot_2023-08-19-15-21-57-70_c3a231c25ed346e59462e84656a70e50](https://github.com/CJNE/ha-myenergi/assets/126982604/2d19e77b-05c2-4c5f-b89e-2ec575df1919)

**Comment by whistlebare** [comment]
![Uploading Screenshot_2023-08-19-15-30-00-49_e4424258c8b8649f6e67d283a50a2cbc.jpg…]()

**Comment by whistlebare** [comment]
![Screenshot_2023-08-19-15-30-00-49_e4424258c8b8649f6e67d283a50a2cbc](https://github.com/CJNE/ha-myenergi/assets/126982604/977f67af-18f6-4daa-b6f9-c6439c79c4a5)

**Comment by whistlebare** [comment]
https://developers.home-assistant.io/docs/core/entity/sensor#long-term-statistics

**Comment by G6EJD** [comment]
I can choose my equivalent of myenergi harvi-10643677 power ct generation in HA.

Have you checked your HA versions and if the ME integration is the latest.

**Comment by whistlebare** [comment]
Yes I keep my Hacs integrations up to date constantly as well as the HA system...

**Comment by G6EJD** [comment]
You can’t add an entity twice, have you copied the energy flow diagram and then added in just the ME entity needed.

**Comment by whistlebare** [comment]
I'm not trying to add it twice as far as I know, I've just listed the two generation entities I have bit neither of these are available to choose from within the solar generation section of the energy tab....only the generated today appears.  My harvi hold that info but it's not able to be chosen in the energy section

**Comment by G6EJD** [comment]
You can only add an entity with units of kWh (Energy) not power, so an instantaneous reading in power is of no use in that design of energy display.

**Comment by whistlebare** [comment]
Well I'm confused then as every other app I have reads live production from solar in kWh...I give up

**Comment by LeiChat** [comment]
Power is Watts and kilo Watts. (kW)
Energy is Watt hours and kilo Watt hours. (kWh)

**Comment by heikone** [comment]
I was also confused, but this example from setup should help to get kW from Myenergi measured values:
`df_myenergi_kW['Solar'] = df_myenergi_sql['gep'] * df_myenergi_sql['v1']/10  /15 / 1000 / 1000`

Measured unit is 15As for 'gep' and 'v1' is V*10 - whyever. I think it's used to keep precision without using decimation.

**Comment by whistlebare** [comment]
> Power is Watts and Kilowatts. (kW) Energy (stored/consumed) is Watt-hours and Kilowatt-hours. (kWh)
> 
> (Electric car motor power is kW. The EV battery capacity is kWh.)
> 
> There are a couple of power flow cards available via HACS. https://github.com/flixlix/power-flow-card-plus https://github.com/ulic75/power-flow-card
> 
> Screenshot of the "power flow card plus" on my dashboard. ![image](https://user-images.githubusercontent.com/23559551/261817113-62c95023-0b84-4b74-ad3c-3ba1f3229d87.png)

Cool that looks like what I'm after, I'll check that out tomorrow

**Comment by whistlebare** [comment]
> I was also confused, but this example from setup should help to get kW from Myenergi measured values: `df_myenergi_kW['Solar'] = df_myenergi_sql['gep'] * df_myenergi_sql['v1']/10 /15 / 1000 / 1000`
> 
> Measured unit is 15As for 'gep' and 'v1' is V*10 - whyever. I think it's used to keep precision without using decimation.

I'm still confused...lol but thanks for trying to explain it

**Comment by G6EJD** [comment]
Both the HA power flow and other power flow cards allow entities to be chosen as required, so no-longer applicable,
